### PR TITLE
Upgrade to React 15.4.1

### PR DESCRIPTION
**Pull request by jacehensley-wf**
## Ultimate Problem:
React 15.4.1 just got release and it came with lots of good stuff.

## Solution:
- Update to React 15.4.1
- Make sure all tests pass

## Testing Suggestions:
- Verify all tests / examples pass

--
FYA: @trentgrover-wf @greglittlefield-wf @clairesarsam-wf @aaronlademann-wf @joelleibow-wf

### CODE REVIEW

**Comment by maxwellpeterson-wf** [comment]
+1

File: lib/react_client.dart [hunk 1]
```diff
@@ -429,7 +429,7 @@ _convertEventHandlers(Map args) {
   args.forEach((propKey, value) {
     var eventFactory = _eventPropKeyToEventFactory[propKey];
     if (eventFactory != null && value != null) {
-      args[propKey] = (events.SyntheticEvent e, [String domId, Event event]) => zone.run(() {
+      args[propKey] = (events.SyntheticEvent e, [_, __]) => zone.run(() {
```
Comment (aaronlademann-wf) [inline lib/react_client.dart]: For my own knowledge @jacehensley-wf - what is the reason for relaxing the type / obfuscating the arguments here?

Comment (jacehensley-wf) [inline lib/react_client.dart]: The reason is because now the second arg passed to this is now of type `EventImpl`. And because we really don't care what it is we can just ignore it.

File: test/js_interop_helpers_test/js_function_test_prod.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <title></title>
 
     <script src="packages/react/react_prod.js"></script>
-    <script src="packages/react/react_dom_prod.js"></script>
+    <script src="packages/react/react_dom.js"></script>
```
Comment (greglittlefield-wf) [inline test/js_interop_helpers_test/js_function_test_prod.html]: Why was this changed?

File: test/index.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <li>
+            <a href="/lifecycle_test.html">
+                Run <code>lifecycle_test.html</code>
+            </a>
+        </li>
+        <li>
+            <a href="/react_test_utils_test.html">
+                Run <code>react_test_utils_test.html</code>
+            </a>
+        </li>
```
Comment (greglittlefield-wf) [inline test/index.html]: Niiiiiiiiice. Would you mind adding links to the `js_interop_helpers_test` files as well, while you're in here?

File: test/js_interop_helpers_test/js_function_test_prod.html (hunk 2 above)
Comment (jacehensley-wf) [inline test/js_interop_helpers_test/js_function_test_prod.html]: Oh, I just changed all `react_dom_prod.js` to `react_dom.js`, this one can be reverted

File: test/index.html (hunk 3 above)
Comment (jacehensley-wf) [inline test/index.html]: Added in 	decc8cd

**Comment by greglittlefield-wf** [comment]
* Changes looks good
* All tests pass ind Dartium and dart2js

+10

**Comment by jacehensley-wf** [comment]
React 15.4.1 just got released, so this PR now bumps to that.

**Comment by aaronlademann-wf** [comment]
+1

**Comment by aaronlademann-wf** [comment]
@trentgrover-wf @greglittlefield-wf this is ready for merge / tag.

**Comment by greglittlefield-wf** [comment]
* Changes still look good
* All tests pass in Dartium and dart2js

+10

Merging
